app/models/project.rb

```diff
@@ -99,12 +120,26 @@ def self.can_edit?(obj, user)
 
     group_ids = user.user_groups.map(&:id)
     obj.projects.each do |project|
+      next if project.open_membership
       return true if group_ids.member?(project.user_group_id) ||
                      group_ids.member?(project.admin_group_id)
     end
     false
   end
 
+  # Check if this user is an admin for a project that includes
+  # this observation.
+  def self.admin_power?(observation, user)
```

Comment:
Question that I think I know the answer to, but worth asking: This method can result in two db queries (`user.user_groups` and `observation.projects`).  This could get ugly really fast if running it on a long list of observations and/or users.  But I'm guessing it's only being used right now by the observation form?  If so, no worries.

Reply:
Yeah, it's only on the observation show page as far as I know.  I've been wondering if there should be some way for admins (or maybe just owners) to download all the location information, but I expect that would go through a different mechanism.